### PR TITLE
Added a function to dump files

### PR DESCRIPTION
**Pull request by Quaker762**
Added a function to dump files, so as to aid in reverse
engineering/inspecting files outside of their respective arc sections.

### CODE REVIEW

**Comment by Quaker762** [comment]
@z33ky This is the new PR, I really cooked the last one lol. Don't drink and git, I guess haha.

File: source/SH3/arc/vfile.cpp [hunk 1]
```diff
@@ -100,3 +102,22 @@ std::size_t sh3_arc_vfile::ReadData(void* destination, std::size_t len, read_err
...
+void sh3_arc_vfile::Dump2Disk()
+{
+    if(!open || buffer.empty())
+    {
+        Log(LogLevel::WARN, "sh3_arc_vfile::Dump2Disk( ): Warning! Attempting to flush unopen or empty buffer to disk!");
+        return;
+    }
+
+    // Firstly, we need to convert the file path to just the filename.
+    std::stringstream ss(fname);
```
Comment (z33ky) [inline source/SH3/arc/vfile.cpp]: unused. remove.
I think you can then also remove the `sstream` include.

File: source/SH3/arc/vfile.cpp [hunk 2]
```diff
@@ -100,3 +102,22 @@ std::size_t sh3_arc_vfile::ReadData(void* destination, std::size_t len, read_err
...
+    std::size_t pos = fname.rfind('/');
+
+    std::ofstream out_file(fname.substr(pos + 1, fname.size()));
+    if(!out_file)
+        return;
+
+    out_file.write((const char*)&buffer[0], buffer.size());
```
Comment (z33ky) [inline source/SH3/arc/vfile.cpp:120]: Doesn't `buffer.data()` (without cast) work for the first arguments?

**Review by z33ky — CHANGES_REQUESTED** [review]

**Comment by z33ky** [comment]
Yeah, rebasing is tricky in the presence of merges. Actually, checkout the documentation for `git-rebase` on `--preserve-merges`. Just a guess that's what happened though, didn't really check out what was going on in your branch there.

File: source/SH3/arc/vfile.cpp (hunk 2 above)
Comment (Quaker762) [reply source/SH3/arc/vfile.cpp:120]: Nah it doesn't, I get an `invalid conversion from unsigned char* to const char*` error

**Review by z33ky — APPROVED** [review]

**Pull request merged**